File: leo2-node/src/main/java/org/deku/leo2/node/rest/services/SwaggerListingResourcePublic.java
package org.deku.leo2.node.rest.services;

import org.deku.leo2.node.rest.swagger.SwaggerContextPublic;
import org.deku.leo2.node.rest.swagger.SwaggerListingResource;

import javax.ws.rs.Path;

/**
 * Delivers swagger.json schema information for public leo2 APIs
 * Created by masc on 20.05.15.
 */
@Path("/")
public class SwaggerListingResourcePublic extends SwaggerListingResource {
    public SwaggerListingResourcePublic() {
        // Inject public swagger context
        super(SwaggerContextPublic.instance());
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/rest/service/SwaggerListingResourcePublic.kt
package org.deku.leoz.node.rest.service

import org.deku.leoz.node.rest.swagger.SwaggerContextPublic
import org.deku.leoz.node.rest.swagger.SwaggerListingResource
import sx.rs.ApiKey

import javax.inject.Named
import javax.ws.rs.Path

/**
 * Delivers swagger.json schema information for public leoz APIs
 * Created by masc on 20.05.15.
 */
@Named
@ApiKey(false)
@Path("/")
class SwaggerListingResourcePublic : SwaggerListingResource(
        // Inject public swagger context
        SwaggerContextPublic)
